test/mlir/conversion/onnx_to_tosa/Math/Elementwise.mlir

```diff
@@ -28,10 +41,47 @@ func.func @test_neg(%arg0: tensor<10x10xf32>) -> tensor<10x10xf32> {
 // CHECK-NEXT:      [[VAR_0_:%.+]] = "tosa.negate"([[PARAM_0_]]) : (tensor<10x10xf32>) -> tensor<10x10xf32>
 }
 
+// -----
 func.func @test_floor(%arg0: tensor<10x10xf32>) -> tensor<10x10xf32> {
   %0 = "onnx.Floor"(%arg0) : (tensor<10x10xf32>) -> tensor<10x10xf32>
   "func.return"(%0) : (tensor<10x10xf32>) -> ()
 // CHECK-LABEL:  func @test_floor
 // CHECK-SAME:   ([[PARAM_0_:%.+]]: tensor<10x10xf32>) -> tensor<10x10xf32> {
 // CHECK-NEXT:      [[VAR_0_:%.+]] = "tosa.floor"([[PARAM_0_]]) : (tensor<10x10xf32>) -> tensor<10x10xf32>
 }
+
+// -----
+func.func @test_add(%arg0: tensor<10x10xf32>, %arg1: tensor<10x10xf32>) -> tensor<10x10xf32> {
+  %0 = "onnx.Add"(%arg0, %arg1) : (tensor<10x10xf32>, tensor<10x10xf32>) -> tensor<10x10xf32>
+  "func.return"(%0) : (tensor<10x10xf32>) -> ()
+// CHECK-LABEL:  func @test_add
+// CHECK-SAME:   ([[PARAM_0_:%.+]]: tensor<10x10xf32>, [[PARAM_1_:%.+]]: tensor<10x10xf32>) -> tensor<10x10xf32> {
+// CHECK-NEXT:      [[VAR_0_:%.+]] = "tosa.add"([[PARAM_0_]], [[PARAM_1_]]) : (tensor<10x10xf32>, tensor<10x10xf32>) -> tensor<10x10xf32>
+}
+
+// -----
+func.func @test_add_int(%arg0: tensor<10x10xi32>, %arg1: tensor<10x10xi32>) -> tensor<10x10xi32> {
+  %0 = "onnx.Add"(%arg0, %arg1) : (tensor<10x10xi32>, tensor<10x10xi32>) -> tensor<10x10xi32>
+  "func.return"(%0) : (tensor<10x10xi32>) -> ()
+// CHECK-LABEL:  func @test_add_int
+// CHECK-SAME:   ([[PARAM_0_:%.+]]: tensor<10x10xi32>, [[PARAM_1_:%.+]]: tensor<10x10xi32>) -> tensor<10x10xi32> {
+// CHECK-NEXT:      [[VAR_0_:%.+]] = "tosa.add"([[PARAM_0_]], [[PARAM_1_]]) : (tensor<10x10xi32>, tensor<10x10xi32>) -> tensor<10x10xi32>
+}
+
+// -----
```

Comment:
I'd be happy to have `test_relu_*` nominated for testing basic infrastructure and not manually add `test_add_int`/`test_sub_int` unless the code implementing the legalization for `fp32` and `i32` differs significantly (which is not the case here).If we wanted coverage of legal input types, I'd prefer more automation around test generation.